unittests/test_loader.py

```diff
@@ -73,9 +74,88 @@ def test_load_bad_init(self):
             'unittests/resources/checks_unlisted/bad_init_check.py')
         assert 0 == len(tests)
 
-    def test_extend_decorator(self):
+    def test_extended_test(self):
         with pytest.warns(ReframeDeprecationWarning) as record:
-            tests = self.loader.load_from_file(
-                'unittests/resources/checks_unlisted/extend_class.py')
-
-        assert len(record) == 3
+            @rfm.simple_test
+            class TestSimple(rfm.RegressionTest):
+                # The test should not raise a deprecation warning even though
+                # it overrides __init__
+                def __init__(self):
+                    pass
```

Comment:
I would only wrap with `pytest.warns()` only those that we expect them to issue a warning.